client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -301,6 +301,50 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerI
     }
 
     @Override
+    public CompletableFuture<Boolean> truncateStream(final String scope, final String stream, final StreamCut streamCut) {
+        return truncateStream(scope, stream, streamCut.getPositions().entrySet()
+                .stream().collect(Collectors.toMap(x -> x.getKey().getSegmentNumber(), Map.Entry::getValue)));
+    }
+
+    @Override
+    public CompletableFuture<Boolean> truncateStream(final String scope, final String stream, final Map<Integer, Long> streamCut) {
```

Comment:
I'm a bit worried about having this API call. It feels like giving a green light for users to go request truncation at arbitrary positions. Why do we actually need it? Could we make it say package protected?

Reply:
its on controller interface that is not exposed to the applications. 
It will be useful for LocalController, though we dont foresee any need for truncation api to be exposed in a local controller as yet. 

i can remove this if you want.